addons/website_event/static/src/js/services/website_event_custom_menus.js

```diff
@@ -0,0 +1,25 @@
+import { registry } from '@web/core/registry';
+import { EditMenuDialog } from '@website/components/dialog/edit_menu';
+
+class CustomEditMenuDialog extends EditMenuDialog {
+    /**
+     * override the addMenu method to handle event page URLs
+     * @override
+     */
+    addMenu(isMegaMenu, customUrl) {
+        let eventPagePrefix = customUrl;
+        const websiteMetadata = this.website.currentWebsite.metadata;
+        if (websiteMetadata.mainObject?.model === "event.event") {
+            const path = new URL(websiteMetadata.path);
+            eventPagePrefix = `/event/${path.pathname.split("/")[2]}/page`;
+        }
+        super.addMenu(isMegaMenu, eventPagePrefix);
```

Comment:
Not sure about this type of magic, it's a bit weird to write your url and then see that we append automatically the event page path to it.

Reply:
Hmm, maybe we can use this `_compute_website_url`? but as we're getting event url in **metadata** which is also needed to check if its an event page, I use that path

Reply:
> perhaps we could prefil the field or have a span before

I'm not sure about this prefill, what if user remove that url part, on creation it will not create an event page.
as for span, don't you think it will look ugly? and users don't really need to see event url, as they just type their page name in url. 
for this like we have add code for `MenuDialog`, `addMenu` and of course make a template.. :thinking:  ?

>Because here, it becomes hard to create a generic page we could add to the event menu and that could be reuse for multiple event (like a condition/privacy page)

hmm, we're creating event page with specific event_id for now, not sure making generic.. :thinking: :thinking: 

Also, this `addmenu` thing is generic so in future if someone needs they can just use this `customUrl` param :wink: 

Thanks! :smile: